### PR TITLE
Feat: add erc721 and erc1155 graphql api

### PR DESCRIPTION
**Pull request by Naupio**
base mainnet branch

### CODE REVIEW

File: lib/godwoken_explorer_web/endpoint.ex [hunk 1]
```diff
@@ -66,10 +66,6 @@ defmodule GodwokenExplorerWeb.Endpoint do
   plug Plug.MethodOverride
   plug Plug.Head
   plug Plug.Session, @session_options
-
-  if Mix.env() in [:stg, :prod] do
-    plug CORSPlug
-  end
-
+  plug CORSPlug
```
Comment (zmcNotafraid) [inline lib/godwoken_explorer_web/endpoint.ex:69]: 这个地方不用修改吧

File: lib/godwoken_indexer/block/sync_worker.ex [hunk 2]
```diff
@@ -725,34 +725,69 @@ defmodule GodwokenIndexer.Block.SyncWorker do
...
+        {map[:address_hash], map[:token_contract_address_hash], map[:block_number],
+         map[:token_id], map[:token_type]}
       end)
+      |> filter_udt_balance_params()
+
+    Import.insert_changes_list(without_token_ids,
+      for: UDTBalance,
+      timestamps: import_utc_timestamps(),
+      on_conflict: :nothing,
+      # conflict_target: [:address_hash, :token_contract_address_hash, :block_number]
```
Comment (zmcNotafraid) [inline lib/godwoken_indexer/block/sync_worker.ex:753]: 注释删掉

File: lib/godwoken_indexer/fetcher/udt_balance.ex [hunk 3]
```diff
@@ -63,42 +69,96 @@ defmodule GodwokenIndexer.Fetcher.UDTBalance do
...
+
+    %{
+      token_contract_address_hash_bytes: token_contract_address_hash.bytes,
+      address_hash_bytes: address_hash.bytes,
+      block_number: block_number,
+      token_id: token_id_int,
+      token_type: token_type
+    }
+
+    # {address_hash.bytes, token_contract_address_hash.bytes, block_number}
```
Comment (zmcNotafraid) [inline lib/godwoken_indexer/fetcher/udt_balance.ex:143]: 注释

File: lib/mix/tasks/update_udt_balance.ex [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+    Repo.all(query)
+    |> Enum.map(fn map ->
+      token_type =
+        if map[:token_type] do
+          map[:token_type]
+        else
+          if map[:token_ids] do
+            :erc1155
+          else
+            if map[:token_id] do
```
Comment (zmcNotafraid) [inline lib/mix/tasks/update_udt_balance.ex:218]: 这里没改啊 TODO

**Review by zmcNotafraid — CHANGES_REQUESTED** [review]

File: lib/godwoken_explorer/graphql/resolovers/token_transfer.ex [hunk 5]
```diff
@@ -13,11 +13,37 @@ defmodule GodwokenExplorer.Graphql.Resolvers.TokenTransfer do
...
+        total_count_primary_key_field: [:transaction_hash, :log_index]
+      })
+
+    {:ok, return}
+  end
+
+  def erc1155_token_transfers(_parent, %{input: input}, _resolution) do
+    return =
+      from(tt in TokenTransfer)
+      |> where([tt], not is_nil(tt.token_ids) or not is_nil(tt.token_id))
```
[the remark this answers is not in the document]
Comment (Naupio) [reply lib/godwoken_explorer/graphql/resolovers/token_transfer.ex:46]: add check  u.eth_type == :erc1155

File: lib/mix/tasks/update_udt_balance.ex (hunk 4 above)
Comment (Naupio) [reply lib/mix/tasks/update_udt_balance.ex:220]: fiexd https://github.com/Magickbase/godwoken_explorer/pull/893/commits/82e2ed050db9b280f625481eded3f1af3f5adf42

**Review by Keith-CY — APPROVED** [review]

**Review by zmcNotafraid — APPROVED** [review]

**Pull request merged**